Local data storage

BenM4k opened the pull request:
We implemented the function that stores the data on the browser, and the one that retrieves them to display them in the form

BenM4k committed:
Update main.js

I added the function that stores the object in the browser

Paul-tes committed:
added JSON parser

BenM4k committed:
Update main.js

fixed eslint errors

Zuhaib042 commented on js/main.js:
```diff
@@ -161,4 +161,29 @@ document.querySelector('#submit').addEventListener('submit', (e) => {
+  localStorage.setItem('form', JSON.stringify(form));
+}
+
+document.querySelector('#submit').addEventListener('submit', storeForm);
```
- [ ] Well done. the data is persisting in the local storage, but it should persist on the input change, when the user changes the content of any input field, not when he submits. so kindly consider refactoring your code to implement this requirement, because right now the data persist on submission and not on input change.
  -  When the user changes the content of any input field, the data is saved to the local storage.

Zuhaib042 reviewed (changes requested):
Hi @BenM4k 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![](https://media.giphy.com/media/3o7qDQAFlF8mZjv6fK/giphy.gif)


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BenM4k committed:
Update main.js

change the event listener to make the data persistant based on the input and not on submit

LionRouge1 reviewed (approved):
Hi @BenM4k, 

# STATUS: APPROVED! 🥇 💯 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

![done](https://media.giphy.com/media/a5llhdWKMOIfneMCpB/giphy.gif)

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
